ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/retriever/RecoveringSidecarRetriever.java

```diff
@@ -106,82 +130,80 @@ public synchronized void stop() {
-    promise.always(() -> pendingPromises.remove(pendingPromiseWithTimestamp));
+  @Override
+  public void onNewValidatedSidecar(final DataColumnSidecar sidecar) {
+    delegate.onNewValidatedSidecar(sidecar);
+  }
 
-    return promise;
+  @VisibleForTesting
+  int pendingRequestsCount() {
+    return pendingRequests.size();
   }
 
-  private void checkPendingPromises() {
+  private void checkPendingRequests() {
     final UInt64 currentTime = timeProvider.getTimeInMillis();
 
-    pendingPromises.removeIf(
-        promiseWithTimestamp -> {
-          if (promiseWithTimestamp.promise.isDone()) {
-            // If the promise is already done, we can remove it
+    pendingRequests.removeIf(
+        requestWithTimestamp -> {
+          if (requestWithTimestamp.response.isDone()) {
+            // If the response is already done, we can remove it
             return true;
           }
           // If the promise is not done, we check if it has timed out
           // TODO-fulu we probably need a better heuristics to submit requests for recovery
           // (https://github.com/Consensys/teku/issues/9465)
-          if (promiseWithTimestamp
+          if (requestWithTimestamp
               .timestamp
-              .plus(recoverInitiationTimeout.toMillis())
+              .plus(recoveryInitiationTimeout.toMillis())
               .isGreaterThan(currentTime)) {
-            // If the promise is not timed out, we keep it
+            // If the response is not timed out, we keep it
             return false;
           }
-
-          // If the promise is timed out, we initiate recovery and remove it
+          // If the response is timed out, we maybe initiate recovery and remove it
           maybeInitiateRecovery(
-              promiseWithTimestamp.dataColumnSlotAndIdentifier, promiseWithTimestamp.promise);
-
+              requestWithTimestamp.dataColumnSlotAndIdentifier, requestWithTimestamp.response);
           return true;
         });
   }
 
-  @Override
-  public void flush() {
-    delegate.flush();
-  }
-
-  @Override
-  public void onNewValidatedSidecar(final DataColumnSidecar sidecar) {
-    delegate.onNewValidatedSidecar(sidecar);
-  }
-
-  @VisibleForTesting
-  int pendingPromisesCount() {
-    return pendingPromises.size();
-  }
-
   private void maybeInitiateRecovery(
-      final DataColumnSlotAndIdentifier columnId, final SafeFuture<DataColumnSidecar> promise) {
+      final DataColumnSlotAndIdentifier columnId, final SafeFuture<DataColumnSidecar> response) {
     blockResolver
         .getBlockAtSlot(columnId.slot())
         .thenPeek(
             maybeBlock -> {
               if (!maybeBlock.map(b -> b.getRoot().equals(columnId.blockRoot())).orElse(false)) {
-                LOG.trace("Recovery: CAN'T initiate recovery for " + columnId);
-                promise.completeExceptionally(
+                LOG.trace("Recovery: CAN'T initiate recovery for {}", columnId);
```

Comment:
this has column and slot i assume coming out? just want to make sure the log has the slot included...